tests/Unit/Models/UserStoryTest.php

```
@@ -445,4 +445,77 @@ function (string $sql, array $params) use ($stmt, &$capturedSql, &$capturedParam
+
+        $result = UserStory::countUnresolvedAlignmentWarningsBatch($db, 99, [7, 8]);
+        $this->assertSame([7 => 2, 8 => 0], $result);
+        $this->assertStringContainsString('alignment_warning IS NOT NULL', $capturedSql);
+        $this->assertStringContainsString('JOIN projects', $capturedSql);
+        $this->assertContains(99, $capturedParams);
+    }
+
+    #[Test]
+    public function countUnresolvedAlignmentWarningsBatchReturnsEmptyForNoProjectIds(): void
+    {
+        $db = $this->createStub(Database::class);
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $db. Configured minimum length is 3. (undefined)

(ShortVariable)